File: Client/src/app/models/dtos/song.dto.ts
import { AlbumDto } from "./album.dto";
import { ArtistDto } from "./artist.dto";
import { GenreDto } from "./genre.dto";
import { TabDto } from "./tab.dto";

export interface SongDto {
    id: string,
    name: string,
    songNumberInAlbum: number,
    fullyMastered: number,
    isFavourite: boolean,
    album: AlbumDto,
    tabs: TabDto[],
    genres: GenreDto[],
    artists: ArtistDto[]
}

File: Tabloid.Domain/DataTransferObjects/SongDto.cs
using System;
using System.Collections.Generic;

namespace Tabloid.Domain.DataTransferObjects
{
    public class SongDto : IDto<Guid>
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public int? SongNumberInAlbum { get; set; }

        public bool FullyMastered { get; set; }

        public bool IsFavourite { get; set; }


        public AlbumDto Album { get; set; }


        public ICollection<TabDto> Tabs { get; set; }

        public ICollection<GenreDto> Genres { get; set; }

        public ICollection<ArtistDto> Artists { get; set; }
    }
}
